src/Analyzer.Core.BuiltInRuleTests/TestTemplates/AppServices.badtemplate

```diff
@@ -10,134 +10,321 @@
+      "dependsOn": [
+        "ApiApp_NoSitesConfig",
+        "WebApp_NoSitesConfig",
+        "FunctionApp_NoSitesConfig"
+      ],
+      "properties": {
+        "cors": {
+          "allowedOrigins": [
+            "*"
+          ]
+        }
+      }
+    },
+    {
+      "apiVersion": "2019-08-01",
+      "type": "Microsoft.Web/sites",
+      "kind": "app",
+      "name": "WebApp_RestrictedCORSAccess_EmbeddedSitesConfig",
+      "location": "[parameters('location')]",
+      "properties": {
+        "httpsOnly": true,
+        "siteConfig": {
+          "cors": {
+            "allowedOrigins": [
+              "someIP"
+            ]
+          }
+        }
+      }
+    },
+    {
+      "apiVersion": "2019-08-01",
+      "type": "Microsoft.Web/sites",
+      "name": "WebApp_NoKind_RestrictedCORSAccess_EmbeddedSitesConfig",
+      "location": "[parameters('location')]",
+      "properties": {
+        "httpsOnly": true,
+        "siteConfig": {
+          "cors": {
+            "allowedOrigins": [
+              "someIP"
+            ]
+          }
+        }
+      }
+    },
+    {
+      "apiVersion": "2019-08-01",
+      "type": "Microsoft.Web/sites",
+      "kind": "app",
+      "name": "WebApp_UnrestrictedCORSAccess_EmbeddedSitesConfig",
+      "location": "[parameters('location')]",
+      "properties": {
+        "httpsOnly": true,
+        "siteConfig": {
+          "cors": {
+            "allowedOrigins": [
+              "someIP",
+              "*"
+            ]
+          }
+        }
+      }
+    },
+    {
+      "apiVersion": "2019-08-01",
+      "type": "Microsoft.Web/sites",
+      "kind": "app",
+      "name": "WebApp_NoSitesConfig",
+      "location": "[parameters('location')]",
+      "properties": {
+        "httpsOnly": true
+      }
+    },
+    {
+      "apiVersion": "2019-08-01",
+      "type": "Microsoft.Web/sites",
+      "kind": "functionapp",
+      "name": "FunctionApp_RestrictedCORSAccess_EmbeddedSitesConfig",
+      "location": "[parameters('location')]",
+      "properties": {
+        "httpsOnly": true,
+        "siteConfig": {
+          "cors": {
+            "allowedOrigins": [
+              "someIP"
+            ]
+          }
+        }
+      }
+    },
+    {
+      "apiVersion": "2019-08-01",
+      "type": "Microsoft.Web/sites",
+      "kind": "functionapp",
+      "name": "FunctionApp_UnrestrictedCORSAccess_EmbeddedSitesConfig",
+      "location": "[parameters('location')]",
+      "properties": {
+        "httpsOnly": true,
+        "siteConfig": {
+          "cors": {
+            "allowedOrigins": [
+              "someIP",
+              "*"
+            ]
+          }
+        }
+      }
+    },
+    {
+      "apiVersion": "2019-08-01",
+      "type": "Microsoft.Web/sites",
+      "kind": "functionapp",
+      "name": "FunctionApp_NoSitesConfig",
+      "location": "[parameters('location')]",
+      "properties": {
+        "httpsOnly": true
+      }
```

Comment:
Can you add a sites/config resource that isn't attached to a resource defined in the template (instead would update a resource already deployed) to verify it doesn't trigger a false positive?

Reply:
If I add this, it would never trigger regardless of the CORS setting since the rule doesn't check stand-alone sites/configs. Can you elaborate on why you think we should add this?

Reply:
Yes, what you've described is the behavior it should validate - i.e. it shouldn't trigger (which would be a false positive if it did).  Without adding that resource, a regression that flags it incorrectly (returning the false positive) might not be caught.

Reply:
@yane3628 I take this back.  Looking at the change, it really wouldn't be a FP if BPA flagged it, we just don't have a great way to do it now.  This can be reverted to how you had it.